runtime/src/main/cpp/Exceptions.cpp

```diff
@@ -206,10 +213,24 @@ void ThrowException(KRef exception) {
 #endif
 }
 
-void ReportUnhandledException(KRef e);
+OBJ_GETTER(Kotlin_setUnhandledExceptionHook, KRef hook) {
+  RETURN_RESULT_OF(SwapRefLocked,
+    &currentUnhandledExceptionHook, currentUnhandledExceptionHook, hook, &currentUnhandledExceptionHookLock);
+}
+
+void OnUnhandledException(KRef throwable) {
+  ObjHolder handlerHolder;
+  auto* handler = SwapRefLocked(&currentUnhandledExceptionHook, currentUnhandledExceptionHook, nullptr,
```

Comment:
This makes it so the exception handler only activates once (i.e. only activates once if you decide in your implementation to rethrow inside the handler)?

Reply:
Rethrowing exception inside the handler would only interrupt the handler itself: https://github.com/JetBrains/kotlin-native/pull/2604/files#diff-e305300924774d54efaaded633649b52R99
Generally the hook is not supposed and can't prevent program termination: https://github.com/JetBrains/kotlin-native/pull/2604/files#diff-f7907f299b4df441dc0eb2cf56d6cbf1R234

Reply:
Ah okay gotcha. The app will continue terminating when this function returns. That's what I was looking for. Thanks!